pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -641,11 +642,21 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd *OSDInfo, provision
 			"",
 		))
 
+	// apply status to each OSD so its key isn't rotated again after a rotation is applied via
+	// daemon restart. important b/c osd update is often interrupted by reconcile requeues
+	cephxStatusVal, err := json.Marshal(osd.CephxStatus)
```

Comment:
Would the generation be sufficient to store on the annotation? Or is the ceph version necessary in addition to the generation? Just asking in the interest of simplifying the annotation to a simple value instead of the json blob.

Reply:
The ceph version will eventually be needed when `WithCephVersionUpdate` is implemented, and it will be beneficial to have it already present if users update and use that feature in the future